ui/lib/core/addon/components/radio-select-ttl-or-string.js

```diff
@@ -0,0 +1,49 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+/**
+ * @module RadioSelectTtlOrString
+ * `RadioSelectTtlOrString` components are yielded out within the formField component when the editType on the model is yield.
+ * The component is two radio buttons, where the first option is a TTL, and the second option is an input field without a title.
+ * This component is used in the PKI engine inside various forms.
+ *
+ * @example
+ * ```js
+ * {{#each @model.fields as |attr|}}
+ *  <RadioSelectTtlOrString @attr={{attr}} @model={{this.model}} />
+ * {{/each}}
+ * ```
+ * @callback onChange
+ * @param {Model} model - Ember Data model that `attr` is defined on.
+ * @param {Object} attr - Usually derived from ember model `attributes` lookup, and all members of `attr.options` are optional.
+ * @param {onChange} [onChange] - Callback triggered on save success.
+ */
+
+export default class RadioSelectTtlOrString extends Component {
+  @tracked groupValue = 'ttl';
+  @tracked ttlTime = '';
+
+  @action selectionChange(selection) {
+    this.groupValue = selection;
+    // Clear the TTL time selection if they have clicked the specificDate radio button.
+    if (selection === 'specificDate') {
+      this.ttlTime = '';
+    }
+  }
+
+  @action setAndBroadcastTtl(value) {
+    let valueToSet = value.enabled === true ? `${value.seconds}s` : 0;
+    this.setAndBroadcast('ttl', `${valueToSet}`);
+  }
+
+  @action setAndBroadcastInput(event) {
+    const prop = event.target.type === 'checkbox' ? 'checked' : 'value';
```

Comment:
Can this line be removed since this is an action in the `<input>` field, the `target.type` will always be 'value' right?

Reply:
Yep! you're right. I was trying to match the pattern from formFields but kept some of the old code.